File: Greedy Algorithm/lemonadeCHANGE.py
def fn( arr ) :
    n = len(arr) 
    five =   ten = twenty = 0

    for i in range(n) :
        if arr[i] == 5 :
            five += 1
        elif arr[i] == 10 :
            if five > 0 : 
                five -= 1 
                ten += 1
            else : 
                return False 
        elif arr[i] == 20 :
            if ten >= 1 and five > 0  :
                twenty += 1 
                ten -+ 1 
                five -= 1 
            elif five >= 3  :
                five -= 3 
            else :
                return False 
    return True


arr = [5,5,5,10,20]
print(fn(arr))
            

File: Greedy Algorithm/lemonadeChange.cpp
#include <bits/stdc++.h>
using namespace std;

bool function1(vector<int> customer)
{
    int n = customer.size();
    int five = 0, ten = 0, twenty = 0;
    for (int i = 0; i < n; i++)
    {
        if (customer[i] == 5)
        {
            five++;
        }
        else if (customer[i] == 10)
        {
            if (five > 0)
            {
                five--;
                ten++;
            }
            else
            {
                return false;
            }
        }
        else if (customer[i] == 20)
        {
            if (ten >= 1 && five > 0)
            {
                twenty++;
                ten--;
                five--;
            }
            else if (five >= 3)
            {
                five -= 3;
                twenty++;
            }
            else{
                return false ;
            }
        }
    }
    return true ;
}


int main()
{
    vector<int> customer = {5, 5, 10, 10, 20};
    bool ans = function1(customer);
    cout << ans;
}